### ISSUE TITLE
Broken bundle when using wildcard and named imports in different places

### ISSUE THREAD

**Issue by shamblesides** [denoland/deno#9560]
# Affected version
Deno 1.7.5

# Input
main.js:
```js
export { foo } from './foo.js';
export { isWater } from "./things.js";
```
foo.js
```js
import * as things from './things.js';

export function foo(x) {
    return things[x]
}
```
things.js
```js
export const water = 'Water';
export function isWater(x) {
    return x === water;
}
```
# Output
`deno bundle main.js`

(comments added to highlight issues)

```js
const mod = function() {
    const water = 'Water';
    const water1 = water; // <-- this is never used
    function isWater(x) {
        return x === water;
    }
    const isWater1 = isWater; // <-- this is never used
    return {
        water: water,
        isWater: isWater
    };
}();
function foo2(x) {
    return mod[x];
}
const foo1 = foo2;
export { foo1 as foo };
export { isWater as isWater }; // <-- Error! isWater is not defined in this scope!
```

**Comment by kitsonk** [comment]
cc/ @kdy1

**Comment by kdy1** [comment]
Fixed on upstream.

**Comment by bartlomieju** [comment]
Fixed in 1.8
